packages/connect/src/api/resetDevice.ts

```diff
@@ -42,10 +48,75 @@ export default class ResetDevice extends AbstractMethod<'resetDevice', PROTO.Res
         };
     }
 
+    // generate host_entropy and then call workflow:
+    // - ResetDevice > EntropyRequest > EntropyAck
+    // - EntropyCheckContinue > EntropyRequest > EntropyAck
+    private async getEntropyData(
+        type: 'ResetDevice' | 'EntropyCheckContinue',
+    ): Promise<EntropyRequestData> {
+        const cmd = this.device.getCommands();
+        const entropy = generateEntropy(32).toString('hex');
+        const params = type === 'ResetDevice' ? this.params : {};
+        const entropyRequest = await cmd.typedCall(type, 'EntropyRequest', params);
+        // EntropyAck > Success if this.params.entropy_check === false
+        await cmd.typedCall('EntropyAck', ['EntropyCheckReady', 'Success'], { entropy });
+
+        return {
+            ...entropyRequest.message,
+            host_entropy: entropy,
+        };
+    }
+
+    private async entropyCheck(prevData: EntropyRequestData): Promise<EntropyRequestData> {
+        const cmd = this.device.getCommands();
+        const paths = ["m/84'/0'/0'", "m/44'/60'/0'"];
+        const xpubs: Record<string, string> = {}; // <path, xpub>
+        for (let i = 0; i < paths.length; i++) {
+            const p = paths[i];
+            const pubKey = await cmd.getPublicKey({ address_n: validatePath(p) });
+            xpubs[p] = pubKey.xpub;
+        }
+
+        const currentData = await this.getEntropyData('EntropyCheckContinue');
+        const res = await verifyEntropy({
+            type: this.params.backup_type,
+            strength: this.params.strength,
+            commitment: prevData.entropy_commitment,
+            hostEntropy: prevData.host_entropy,
+            trezorEntropy: currentData.prev_entropy,
+            xpubs,
+        });
+        if (res.error) {
+            throw ERRORS.TypedError('Failure_EntropyCheck', res.error);
+        }
+
+        return currentData;
+    }
+
     async run() {
         const cmd = this.device.getCommands();
-        const response = await cmd.typedCall('ResetDevice', 'Success', this.params);
```

Comment:
Different expected response from `ResetDevice` (`Success` before, `EntropyRequest` now) is because previously `ResetDevice` always returned `EntropyRequest`, which was acked from DeviceCommands which returned `Success`? 

I just want to make sure that there's no change in message flow apart from extending it.

Reply:
FYI, if it's any help the workflow is documented [here](https://github.com/trezor/trezor-firmware/blob/main/docs/common/message-workflows.md#resetdevice).

Reply:
Thanks, I know about this one. It's just that from our code it seemed that `ResetDevice` previously returned `Success`, but now I believe it was just because `EntropyRequest`/`EntropyAck` were hidden under the hood.

Reply:
well, not exactly
previous flow also returns EntropyRequest but it was handled by DeviceCommands._filterCommonTypes as we didnt work with EntropyRequest data until now. same flow is also used if `entropy_check` is set to false 

(Host) ResetDevice > (Trezor) EntropyRequest > (Host) EntropyAck > (Trezor) Success